pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,195 @@
+package latencyprofilecontroller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+
+	apiconfigv1 "github.com/openshift/api/config/v1"
+	configv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	listerv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	nodeobserver "github.com/openshift/library-go/pkg/operator/configobserver/node"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	revisionConfigMapName = "config"
+	revisionConfigMapKey  = "config.yaml"
+)
+
+// LatencyProfileController either instantly via the informers
+// or periodically via resync, lists the config/v1/node object
+// and fetches the worker latency profile applied on the cluster which is used to
+// updates the status of respective operator resource that uses this controller.
+// The current state of the operand is by watching the configs applied
+// to different static pod revisions that are active and uses the information to
+// update status as either progressing or completed or degraded.
+
+type LatencyProfileController struct {
+	operatorClient          v1helpers.StaticPodOperatorClient
+	targetNamespace         string
+	configClient            configv1.ConfigV1Interface
+	configMapClient         corev1client.ConfigMapsGetter
+	configNodeLister        listerv1.NodeLister
+	latencyProfileArgValMap map[apiconfigv1.WorkerLatencyProfileType][]nodeobserver.LatencyProfileArgVal
+}
+
+func NewLatencyProfileController(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	targetNamespace string,
+	latencyProfileArgValMap map[apiconfigv1.WorkerLatencyProfileType][]nodeobserver.LatencyProfileArgVal,
+	configClient configv1.ConfigV1Interface,
```

Comment:
why do we need this client?

Reply:
Adding that client was erroneous/extra/left back when I copied over old code. 
Removed it now.